### ISSUE TITLE
vue template script buggy

### ISSUE THREAD

**Issue by ahkui** [vuejs/create-vue#472]
1. create new project with `npm create vue@latest`
2. build static file with `.env.ci` file

the second step i have try under command:
- `npm run build -- --mode ci` (not working)
  <img width="443" alt="image" src="https://github.com/vuejs/create-vue/assets/14049597/88e001f4-9b1f-4f00-8bea-9257f715d3f3">
- `npm run build-only -- --mode ci` (working)
  <img width="436" alt="image" src="https://github.com/vuejs/create-vue/assets/14049597/af995fec-5130-47ae-afd5-3bcc9fc469fd">
- `npm run build -- -- --mode ci` (working)
  <img width="447" alt="image" src="https://github.com/vuejs/create-vue/assets/14049597/e0853cdb-52bc-4a8a-b96a-b766b950fcb1">

from this [document](https://github.com/mysticatea/npm-run-all/blob/master/docs/run-p.md#argument-placeholders), i think the build script need do some change

`"build": "run-p type-check \"build-only {@}\" --"` change to `"build": "run-p type-check \"build-only -- {@}\" --"`
<img width="716" alt="image" src="https://github.com/vuejs/create-vue/assets/14049597/5baf9840-b063-4377-865d-2ba0bc05e8e6">

**Comment by Yordan-Ramchev** [comment]
Different package managers (npm, yarn, pnpm & bun) are not handling CLI the same way.
#### Related:
1. https://github.com/vuejs/create-vue/issues/393
2. https://github.com/vuejs/create-vue/pull/394

**Comment by cexbrayat** [comment]
Yes, as @Yordan-Ramchev pointed out, this is a duplicate of #393 
Let's track the older one and close this one.
